### PR TITLE
Evaluation Schema + a necessary revamp

### PR DESCRIPTION
**Pull request by Udayraj123**
Fixes https://github.com/Udayraj123/OMRChecker/issues/89 
This PR attempts to provide a generic solution for evaluation.

This is PR#1 of a chained PR. 
[Next](https://github.com/Udayraj123/OMRChecker/pull/124)

### CODE REVIEW

**Comment by Udayraj123** [comment]
@Rohan-G you can start having a look in the main files. Starting with the schema. I'll push working changes soon.

File: samples/community/Sandeep-1507/template.json [hunk 1]
```diff
@@ -1,7 +1,5 @@
 {
   "Globals": {
-    "window_width": 1189,
```
Comment (deepakgouda) [inline samples/community/Sandeep-1507/template.json:3]: Sorry if I'm unaware of the naming convention, but are we keeping the folder name according to contributors or moving to something generic/mainstream.

Comment (Udayraj123) [reply samples/community/Sandeep-1507/template.json:3]: Inside community folder, we're allowing credits to the ones who created the sample template json. This is to incentivise new OMR templates getting submitted and also get new users onboarded for issue https://github.com/Udayraj123/OMRChecker/issues/40

**Comment by Udayraj123** [comment]
The changes include(wip list):

working evaluation code
add range operator support
add validations for marking scheme and omr response
picking up window_width (using screeninfo)
pickup evaluation.json recursively;
added schema for all 3 jsons
improved error messages
reduce hardcoded configs
support config.json per instance
pickup config.json recursively;
test threshold jump value
upgrade pip packages
load json error handling
separate out utils into ImageInstanceOps, ImageUtils(static methods) and InteractionUtils(static methods)
add upsc mock; refactor; create evaluation-schema

**Comment by Udayraj123** [comment]
The changes include(wip list):
- working evaluation code
- add range operator support
- add validations for marking scheme and omr response
- picking up window_width (using screeninfo)
- pickup evaluation.json recursively;
- added schema for all 3 jsons
- improved error messages
- reduce hardcoded configs
- support config.json per instance
- pickup config.json recursively;
- test threshold jump value
- upgrade pip packages
- load json error handling
- separate out utils into ImageInstanceOps, ImageUtils(static methods) and InteractionUtils(static methods)
- add upsc mock; refactor; create evaluation-schema

**Review by deepakgouda — APPROVED** [review]

File: src/evaluation.py [hunk 2]
```diff
@@ -0,0 +1,598 @@
+import ast
```
Comment (ayushjain2809) [inline src/evaluation.py:1]: A couple of articles that might be useful for refactoring this file:
https://dzone.com/articles/the-stepdown-rule
https://dzone.com/articles/slap-your-methods-and-dont-make-me-think

File: src/logger.py [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 from typing import Union
 
+from rich.console import Console
 from rich.logging import RichHandler
 
-from .config import CONFIG_DEFAULTS as config
-
 FORMAT = "%(message)s"
 
+# TODO: set logging level from config.json dynamically
```
[the remark this answers is not in the document]
Comment (Udayraj123) [reply src/logger.py:9]: will add in a later PR

File: src/evaluation.py [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+)
+from src.utils.parsing import get_concatenated_response, open_evaluation_with_validation
+
+
+def parse_float_or_fraction(result):
+    if type(result) == str and "/" in result:
+        result = float(Fraction(result))
+    else:
+        result = float(result)
+    return result
```
[the remark this answers is not in the document]
Comment (Udayraj123) [reply src/evaluation.py:26]: addressing this in PR#3 now

File: src/evaluation.py (hunk 2 above)
Comment (Udayraj123) [reply src/evaluation.py:1]: Will do via PR#3

File: src/evaluation.py [hunk 5]
```diff
@@ -0,0 +1,598 @@
...
+from rich.table import Table
+
+from src.logger import console, logger
+from src.schemas.evaluation_schema import (
+    BONUS_SECTION_PREFIX,
+    DEFAULT_SECTION_KEY,
+    MARKING_VERDICT_TYPES,
+    QUESTION_STRING_REGEX_GROUPS,
+)
+from src.utils.parsing import get_concatenated_response, open_evaluation_with_validation
```
[the remark this answers is not in the document]
Comment (Udayraj123) [reply src/evaluation.py]: It seems like isort hook and black standards is keeping the same order as given. Moving `import cv2` to top again moves it back to where it was

**Pull request merged**